plugins/opensslcryptoplugin/cryptoplugin_common.cpp

```diff
@@ -407,15 +463,51 @@ CRYPTOPLUGINCOMMON_NAMESPACE::CRYPTOPLUGINCOMMON_CLASS::verify(
         Sailfish::Crypto::CryptoManager::DigestFunction digestFunction,
         bool *verified)
 {
-    // TODO: support more operations and algorithms in this plugin!
-    Q_UNUSED(signature);
-    Q_UNUSED(data);
-    Q_UNUSED(key);
-    Q_UNUSED(padding);
-    Q_UNUSED(digestFunction);
-    Q_UNUSED(verified);
-    return Sailfish::Crypto::Result(Sailfish::Crypto::Result::UnsupportedOperation,
-                                    QLatin1String("TODO: verify"));
+    if (padding != Sailfish::Crypto::CryptoManager::SignaturePaddingNone) {
+        return Sailfish::Crypto::Result(Sailfish::Crypto::Result::UnsupportedOperation,
+                                        QLatin1String("TODO: signature padding other than None"));
+    }
+
+    QScopedPointer<BIO, LibCrypto_BIO_Deleter> bio(BIO_new(BIO_s_mem()));
+    QScopedPointer<EVP_PKEY, LibCrypto_EVP_PKEY_Deleter> pkey(EVP_PKEY_new());
+
+    int r;
+    *verified = false;
+
+    // Use BIO to write public key data to bio
+    r = BIO_write(bio.data(), key.publicKey().data(), key.publicKey().length());
+    if (r != 1) {
+        return Sailfish::Crypto::Result(Sailfish::Crypto::Result::CryptoPluginSigningError,
+                                        QLatin1String("Failed to read private key data."));
+    }
+
+    // Read the public key data into an EVP_PKEY
+    EVP_PKEY *pkeyPtr = pkey.data();
+    PEM_read_bio_PUBKEY(bio.data(), &pkeyPtr, Q_NULLPTR, Q_NULLPTR);
+
+    // Get the EVP digest function
+    const EVP_MD *evpDigestFunc = getEvpDigestFunction(digestFunction);
+    if (!evpDigestFunc) {
+        return Sailfish::Crypto::Result(Sailfish::Crypto::Result::UnsupportedDigest,
+                                        QLatin1String("Unsupported digest function chosen."));
+    }
+
+    // Verify the signature
+    r = osslevp_verify(evpDigestFunc, pkeyPtr, data.data(), data.length(), (const uint8_t*) signature.data(), (size_t) signature.length());
+    if (r == 1) {
+        // Verified successfully.
+        *verified = true;
+        return Sailfish::Crypto::Result(Sailfish::Crypto::Result::Succeeded);
+    }
```

Comment:
Sailfish OS coding style https://sailfishos.org/wiki/Coding_Conventions requires the else on the same line as the closing-brace of the if block, i.e. } else if (...) {

Reply:
Sorry, I'm not used to the TIE-fighter else, but will change it!